example/test/avoid_unnecessary_type_assertions_test.dart

```diff
@@ -0,0 +1,33 @@
+// ignore_for_file: prefer_const_declarations
+// ignore_for_file: unnecessary_nullable_for_final_variable_declarations
+// ignore_for_file: unnecessary_type_check
+// ignore_for_file: unused_local_variable
+
+/// Check the `avoid-unnecessary-type-assertions` rule
+
+void fun() {
+  final testList = <double>[1.0, 2.0, 3.0];
+  // expect_lint: avoid-unnecessary-type-assertions
+  final result = testList is List<double>;
```

Comment:
Not sure we need to be explicit here. Let type inference do its job.

```suggestion
  final testList = [1.0, 2.0, 3.0];
  // expect_lint: avoid-unnecessary-type-assertions
  final result = testList is List<double>;
```